src/render.c

```diff
@@ -305,13 +305,32 @@ void render_close(void) { render_ingame_cleanup(); }
  *              the COLOR_PAIR macro
  */
 static void render_cell(WINDOW *win, int row, int col, short color) {
-	for (int i = 1; i <= CELL_WIDTH; i++)
-		mvwaddch(win, 1 + row, i + col * CELL_WIDTH, ' ');
-	// Handle Shadow Block
-	if (color == shadow)
-		color = 8;
-	mvwchgat(win, 1 + row, 1 + col * CELL_WIDTH, CELL_WIDTH, 0, color,
-	         NULL);
+	int i;
+	// Reset Cell
+	mvwchgat(win, 1 + row, 1 + col * CELL_WIDTH, CELL_WIDTH, 0, -1, NULL);
```

Comment:
Nit: I believe you can move this statement into the `color == shadow` conditional since this there is a corresponding `mvchgat` call on [line 331](https://github.com/Nitepone/tetris-mint/pull/51/files#diff-2d4534614d8fddc6fa5aa4a49a27a5112bd7116cd648fd3ede43193490279418R331) for non-shadow blocks.

Reply:
Fixed in new patch. I was indecisive on the correct solution here.

I was concerned about if we were to add more special render cases. Having the reset happen no matter what could save headache. That said, its a waste of resources to do every time.